FF1RandomizerOnline/wwwroot/js/site.js

```diff
@@ -146,56 +151,65 @@ function forcedPartyMembersCallback() {
 }
 
 function getFlagsString() {
-	var checkboxBits = 0;
+	// We put all the information into an array of bools and work it into a tight base64 string later.
+	var flagBools = [];
 	for (var i = 0; i < checkboxIds.length; i++) {
 		var checkbox = document.getElementById(checkboxIds[i]);
-		if (checkbox.checked) {
-			checkboxBits |= 1 << i;
-		}
+		flagBools.push(!!checkbox.checked);
 	}
+
 	var select = document.getElementById("Flags_TeamSteak");
-	if (select.value === "True") {
-		checkboxBits |= 1 << checkboxIds.length;
-	}
+	flagBools.push(select.value === "True");
 
 	select = document.getElementById("Flags_Music");
-	if (select.value === "Standard") {
-		checkboxBits |= 1 << (checkboxIds.length + 1);
+	if (select.value === "None") {
+		flagBools.push(false);
+		flagBools.push(false);
+	} else if (select.value === "Standard") {
+		flagBools.push(true);
+		flagBools.push(false);
 	} else if (select.value === "Nonsensical") {
-		checkboxBits |= 1 << (checkboxIds.length + 2);
+		flagBools.push(false);
+		flagBools.push(true);
 	} else if (select.value === "MusicDisabled") {
-		checkboxBits |= 1 << (checkboxIds.length + 1);
-		checkboxBits |= 1 << (checkboxIds.length + 2);		
+		flagBools.push(true);
+		flagBools.push(true);
+	}
+
+	// Put the bool array into a uint8 array. We read beyond the
+	// end of flagBools here but it just gives undefineds which become 0s.
+	// If the order of the bits is odd it's because we match C#'s bits.CopyTo
+	var flagBytes = new Uint8Array(Math.ceil((flagBools.length + 0.0) / 8));
+	for (var i = 0; i < flagBytes.length; ++i) {
+		flagBytes[i] = (
+			((flagBools[i * 8 + 0] ? 0x01 : 0x00) << 0) |
+			((flagBools[i * 8 + 1] ? 0x01 : 0x00) << 1) |
+			((flagBools[i * 8 + 2] ? 0x01 : 0x00) << 2) |
+			((flagBools[i * 8 + 3] ? 0x01 : 0x00) << 3) |
+			((flagBools[i * 8 + 4] ? 0x01 : 0x00) << 4) |
+			((flagBools[i * 8 + 5] ? 0x01 : 0x00) << 5) |
+			((flagBools[i * 8 + 6] ? 0x01 : 0x00) << 6) |
+			((flagBools[i * 8 + 7] ? 0x01 : 0x00) << 7)
+		);
 	}
 
-	var flagsString = "";
-	var charBits;
-	charBits = (checkboxBits & 0x000000FC) >>>  2;
-	flagsString += base64Chars[charBits];
-	charBits = (checkboxBits & 0x00000003) <<   4 | (checkboxBits & 0x0000F000) >>> 12;
-	flagsString += base64Chars[charBits];
-	charBits = (checkboxBits & 0x00000F00) >>>  6 | (checkboxBits & 0x00C00000) >>> 22;
-	flagsString += base64Chars[charBits];
-	charBits = (checkboxBits & 0x003F0000) >>> 16;
-	flagsString += base64Chars[charBits];
-	charBits = (checkboxBits & 0xFC000000) >>> 26;
-	flagsString += base64Chars[charBits];
-	charBits = (checkboxBits & 0x03000000) >>> 20;
-	flagsString += base64Chars[charBits];
+	var flagsString = btoa(String.fromCharCode.apply(null, flagBytes));
```

Comment:
All the work of this patch is to leverage btoa here and atob later.
The javascript string replaces use regexes so they behave like replaceAll